library/common/jni/android_jni_interface.cc

```diff
@@ -15,6 +16,13 @@ Java_io_envoyproxy_envoymobile_engine_AndroidJniLibrary_initialize(JNIEnv* env,
                                                                    jobject class_loader,
                                                                    jobject connectivity_manager) {
   set_class_loader(env->NewGlobalRef(class_loader));
+  // At this point, we know Android APIs are available. Register cert chain validation JNI calls.
+  envoy_status_t result =
+      register_platform_api(cert_validator_name, get_android_cert_validator_api());
+  if (result == ENVOY_FAILURE) {
+    return -1;
```

Comment:
I wonder whether we should just `return register_platform_api...` in here? `ENVOY_SUCCESS` is equal to `1` which is different than -1 that we return in here but I guess we decided to use ENVOY_SUCCESS/ENVOY_FAILURE constants so shiould be probably fine to return to simplify this and do `return register_platform_api...`

Thoughts?

Reply:
discussed offline - looks good the way it is.

Reply:
Changed to return `ENVOY_FAILURE` here and made `android_test_jni_inteface.cc` return directly.